1) A #define in xproto breaks the KDE headers
2) gid_t/uid_t are typedef'd unsigned short

--- origsrc/kde-workspace-4.11.18/kcontrol/dateandtime/helper.cpp	2015-04-08 02:30:11.000000000 -0500
+++ src/kde-workspace-4.11.18/kcontrol/dateandtime/helper.cpp	2015-05-07 10:49:40.366355300 -0500
@@ -25,6 +25,7 @@
 
 */
 
+#define _BSD_SOURCE
 #include "helper.h"
 
 #include <config-workspace.h>
--- origsrc/kdebase-workspace-4.3.4/kcontrol/input/main.cpp	2008-01-04 17:55:14.000000000 -0600
+++ src/kdebase-workspace-4.3.4/kcontrol/input/main.cpp	2010-01-15 03:11:06.439852500 -0600
@@ -26,14 +26,13 @@
 #include <klocale.h>
 #include <kglobal.h>
 #include <kconfig.h>
+#include <ktoolinvocation.h>
+#include <klauncher_iface.h>
 #include <QFile>
 
 #include "mouse.h"
 #include <QX11Info>
 
-#include <ktoolinvocation.h>
-#include <klauncher_iface.h>
-
 #include <X11/Xlib.h>
 #ifdef HAVE_XCURSOR
 #  include <X11/Xcursor/Xcursor.h>
--- origsrc/kdebase-workspace-4.4.2/kcontrol/kfontinst/dbus/FontInst.cpp	2010-01-31 13:30:18.000000000 -0600
+++ src/kdebase-workspace-4.4.2/kcontrol/kfontinst/dbus/FontInst.cpp	2010-05-04 11:51:21.895650400 -0500
@@ -391,8 +391,8 @@ void FontInst::move(const QString &famil
             args["files"] = files;
             args["toSystem"] = toSystem;
             args["dest"] = theFolders[to].location();
-            args["uid"] = getuid();
-            args["gid"] = getgid();
+            args["uid"] = uint(getuid());
+            args["gid"] = uint(getgid());
             int result=performAction(args);
 
             if(STATUS_OK==result)
